editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-property-changes.ts

```diff
@@ -215,27 +221,92 @@ export function getStaticReparentPropertyChanges(
 }
 
 export function positionElementToCoordinatesCommands(
-  elementPath: ElementPath,
+  elementToReparent: ElementPathSnapshots,
+  oldAllElementProps: AllElementProps,
+  metadata: MetadataSnapshots,
   desiredTopLeft: CanvasPoint,
+  pathLookup: OldPathToNewPathMapping,
 ): CanvasCommand[] {
-  return [
-    ...pruneFlexPropsCommands(flexChildProps, elementPath),
+  const basicCommands = [
+    ...pruneFlexPropsCommands(flexChildProps, elementToReparent.newPath),
     setCssLengthProperty(
       'always',
-      elementPath,
+      elementToReparent.newPath,
       PP.create('style', 'top'),
       { type: 'EXPLICIT_CSS_NUMBER', value: cssNumber(desiredTopLeft.y, null) },
       null,
     ),
     setCssLengthProperty(
       'always',
-      elementPath,
+      elementToReparent.newPath,
       PP.create('style', 'left'),
       { type: 'EXPLICIT_CSS_NUMBER', value: cssNumber(desiredTopLeft.x, null) },
       null,
     ),
-    setProperty('always', elementPath, PP.create('style', 'position'), 'absolute'),
+    setProperty('always', elementToReparent.newPath, PP.create('style', 'position'), 'absolute'),
   ]
+
+  const elementIsFragmentLike = treatElementAsFragmentLike(
+    metadata.originalTargetMetadata,
+    oldAllElementProps,
+    metadata.originalPathTrees,
+    elementToReparent.oldPath,
+  )
+
+  if (!elementIsFragmentLike) {
+    return basicCommands
+  }
+
+  const children = replaceFragmentLikePathsWithTheirChildrenRecursive(
+    metadata.originalTargetMetadata,
+    oldAllElementProps,
+    metadata.originalPathTrees,
+    [elementToReparent.oldPath],
+  )
+
+  const containerBounds = MetadataUtils.getFrameInCanvasCoords(
+    elementToReparent.oldPath,
+    metadata.originalTargetMetadata,
+  )
+
+  if (containerBounds == null || isInfinityRectangle(containerBounds)) {
+    return basicCommands
+  }
+
+  const childrenPositioningCommands = children.flatMap((child) => {
+    const childBounds = MetadataUtils.getFrameInCanvasCoords(child, metadata.originalTargetMetadata)
+    if (childBounds == null || isInfinityRectangle(childBounds)) {
+      return []
+    }
+    const adjustedTop = desiredTopLeft.y + childBounds.y - containerBounds.y
```

Comment:
this seems odd to me that you need to adjust the offset here, if this is a child of Fragment then the same coordinate system "rules" should apply that were applicable when you created `basicCommands`.

Reply:
This is because `desiredTopLeft` is calculated based on the bounds of the fragment-like element. Without this adjustment, the offsets of the children would stay the same, putting them at a wildly wrong offset. This adjustment does the same thing that multiselect paste does, positioning the children so that their bounding box (the BB of their parent fragment/conditional) will be at `desiredTopLeft`, with the relative offset of the children preserved.

Reply:
ah I see